types/json_printing.h

```diff
@@ -186,9 +186,12 @@ struct StreamJsonPrintingContext
 
     virtual void writeString(const std::string & s)
     {
+        writeStringUtf8(Utf8String(s));
```

Comment:
No, std::string should have Ascii.  We should add a writeUtf8String(const Utf8String &) method and make sure we call it from Id.